algorithms/src/snark/marlin/data_structures/circuit_verifying_key.rs

```diff
@@ -50,9 +51,10 @@ pub struct CircuitVerifyingKey<E: PairingEngine, MM: MarlinMode> {
     pub verifier_key: sonic_pc::VerifierKey<E>,
     #[doc(hidden)]
     pub mode: PhantomData<MM>,
+    pub id: CircuitId,
 }
 
-impl<E: PairingEngine, MM: MarlinMode> Prepare for CircuitVerifyingKey<E, MM> {
+impl<E: PairingEngine, MM: MarlinMode> PrepareOrd for CircuitVerifyingKey<E, MM> {
```

Comment:
Why is this `PrepareOrd`?

Reply:
For context:
```
/// Defines trait that describes preparing from an unprepared version to a prepare version.
pub trait Prepare {
    type Prepared;
    fn prepare(&self) -> Self::Prepared;
}

/// Defines trait that describes preparing from an unprepared version to an orderable prepare version.
pub trait PrepareOrd {
    // NOTE: we keep this separate from Prepare because we also have unordered Prepared types
    type Prepared: Ord;
    fn prepare(&self) -> Self::Prepared;
}
```

So it simply makes sure that Prepared types have an ordering, which is useful now that we need to use `CircuitVerifyingKey` as ordered key in `keys_to_inputs` (which is passed to `verify_batch_prepared(...)`.

I wanted to keep around the original `Prepare` trait, which is used by `PreparedCommitment` and `PreparedVerifierKey` which don't need to define an ordering.